src/cli/cli.hpp

```diff
@@ -531,10 +533,31 @@ class Interpreter : public OutputImplementer, public Output
     static void HandleTimer(Timer &aTimer);
     void        HandleTimer(void);
 
-    const otCliCommand *mUserCommands;
-    uint8_t             mUserCommandsLength;
-    void               *mUserCommandsContext;
-    bool                mCommandIsPending;
+    class otCliCommandList : public LinkedListEntry<otCliCommandList>
```

Comment:
Remove the `ot` prefix from the name.

`ot` prefix should be used for OT public C definitions. This is C++ and we are already under namespace `ot`.